pre_commit/languages/haskell.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import contextlib
+import logging
+import pathlib
+import subprocess
+import typing
+
+import pre_commit.envcontext as _envcontext
+import pre_commit.lang_base as _lang_base
+import pre_commit.prefix as _prefix
+
+_LOGGER = logging.getLogger(__name__)
+
+ENVIRONMENT_DIR = 'hs_env'
+
+get_default_version = _lang_base.basic_get_default_version
+health_check = _lang_base.basic_health_check
+run_hook = _lang_base.basic_run_hook
+
+
+@contextlib.contextmanager
+def in_env(prefix: _prefix.Prefix, version: str) -> typing.Iterator[None]:
+    environment_dir = _lang_base.environment_dir(
+        prefix,
+        ENVIRONMENT_DIR,
+        version,
+    )
+    _LOGGER.debug(f'environment_dir: {environment_dir}')
+
+    with _envcontext.envcontext(patch=()):
+        yield
+
+
+def install_environment(
+    prefix: _prefix.Prefix,
+    version: str,
+    additional_dependencies: typing.Sequence[str],
+) -> None:
+    environment_dir = pathlib.Path(
+        _lang_base.environment_dir(prefix, ENVIRONMENT_DIR, version),
+    )
+    print(f'environment_dir: {environment_dir}')
+    _LOGGER.debug(f'environment_dir: {environment_dir}')
+    print(f'pathlib.Path.cwd(): {pathlib.Path.cwd()}')
+    _LOGGER.debug(f'pathlib.Path.cwd(): {pathlib.Path.cwd()}')
+
+    cabal_files = [str(path) for path in pathlib.Path.cwd().glob('*.cabal')]
+    assert len(cabal_files) == 1, (
+        f"didn't find a unique cabal file: {', '.join(cabal_files)}"
+    )
+
+    subprocess.run(
```

Comment:
there shouldn't be any usage of `subprocess` directly -- these should go through `setup_cmd` so they have the proper location -- this is probably most of your issues with getting this to work

Reply:
That sounds like the winner.  I'll give that a shot and see what happens.  Thanks.